Engine/Graphics/OpenGL/RenderOpenGL.cpp

```diff
@@ -2434,18 +2434,26 @@ void RenderOpenGL::RemoveTextureFromDevice(Texture* texture) {
 
 bool RenderOpenGL::MoveTextureToDevice(Texture *texture) {
     auto t = (TextureOpenGL *)texture;
-    auto native_format = t->GetFormat();
-    int gl_format = GL_RGB;
-        // native_format == IMAGE_FORMAT_A1R5G5B5 ? GL_RGBA : GL_RGB;
+    IMAGE_FORMAT native_format = t->GetFormat();
 
     unsigned __int8 *pixels = nullptr;
-    if (native_format == IMAGE_FORMAT_R5G6B5 || native_format == IMAGE_FORMAT_A1R5G5B5 || native_format == IMAGE_FORMAT_A8R8G8B8 || native_format == IMAGE_FORMAT_R8G8B8A8) {
+    int gl_format = 0;
+    switch (native_format) {
+    case IMAGE_FORMAT_R5G6B5:
+    case IMAGE_FORMAT_A1R5G5B5:
+    case IMAGE_FORMAT_A8R8G8B8:
+    case IMAGE_FORMAT_R8G8B8A8:
         pixels = (unsigned __int8 *)t->GetPixels(IMAGE_FORMAT_A8R8G8B8);
-        // takes care of endian flip from literals here - hence BGRA
-        gl_format = GL_BGRA;
-    } else {
+        gl_format = GL_BGRA; // takes care of endian flip from literals here - hence BGRA
+        break;
+    case IMAGE_FORMAT_R8G8B8:
+        pixels = (unsigned __int8 *)t->GetPixels(IMAGE_FORMAT_R8G8B8);
+        gl_format = GL_RGB;
+        break;
```

Comment:
Please drop this. Consistent use of GL_BGRA is preferred to avoid any GL packing issues.

Reply:
Dropped.